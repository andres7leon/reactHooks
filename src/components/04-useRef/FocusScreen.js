import React, { useRef } from 'react'

export const FocusScreen = () => {


    const inputRef = useRef()

    const handleClick = () => {
        inputRef.current.select();
    }

    return (
        <div>
            <h1>Focus screen</h1>

            <hr/>

            <input 
                ref={inputRef}
                className='form-control'
                placeholder='su nombre'
            />

            <button className='btn btn-info' onClick={ handleClick }>Focus</button>
        </div>
    )
}
