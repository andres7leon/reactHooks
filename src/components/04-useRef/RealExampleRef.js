import React, { useState } from 'react'
import { MultiplesCustomshooks } from '../03-examples/MultiplesCustomshooks'

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1> RealExpample Ref</h1>
            <hr/>

            { show && <MultiplesCustomshooks />}

            <button
                className='btn btn-info mt-3'
                onClick={ () => {
                    setShow( !show )
                }}
                >
                    Show - hide
                </button>

        </div>
    )
}
