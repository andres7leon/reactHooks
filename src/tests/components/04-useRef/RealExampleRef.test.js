import React from 'react';
import {shallow} from 'enzyme'
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('pruebas hook RealExampleRef', () => {
   
    const wrapper = shallow(<RealExampleRef />);

    test('debe de mostrarse bien ', () => {
       
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultiplesCustomshooks').exists() ).toBe(false);
        
    });
    
    test('debe de mostrar el compoente <multiplescustom', () => {
        
        wrapper.find('button').simulate('click');

        expect( wrapper.find('MultiplesCustomshooks').exists() ).toBe(true);
        
    });

});
