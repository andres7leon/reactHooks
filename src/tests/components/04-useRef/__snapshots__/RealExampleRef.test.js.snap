// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas hook RealExampleRef debe de mostrarse bien  1`] = `
<div>
  <h1>
     RealExpample Ref
  </h1>
  <hr />
  <button
    className="btn btn-info mt-3"
    onClick={[Function]}
  >
    Show - hide
  </button>
</div>
`;
